### PR TITLE
[Example] 3D-Unet

### PR DESCRIPTION
**Pull request by cagery**
Provides 3D image segmentation inference example using brats dataset. Based on mlcommons ONNX model.

### CODE REVIEW

File: examples/python_3dunet/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# 3D-Unet Inference with AMD MIGraphX
+
+This example provides how to apply image segmentation to 3D images using AMD MIGraphX on a given AMD GPU. 
+
+## How to:
+1) User will need to have access to BRATS dataset.
```
Comment (turneram) [inline examples/python_3dunet/README.md:6]: It would be helpful to give more detailed instructions for how to access the dataset.

Comment (cagery) [reply examples/python_3dunet/README.md]: added the link for getting access

File: examples/python_3dunet/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# 3D-Unet Inference with AMD MIGraphX
+
+This example provides how to apply image segmentation to 3D images using AMD MIGraphX on a given AMD GPU.
```
Comment (kahmed10) [inline examples/python_3dunet/README.md:3]:
```suggestion
This example applies image segmentation to 3D images using AMD MIGraphX on a given AMD GPU. 
```

File: examples/python_3dunet/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# 3D-Unet Inference with AMD MIGraphX
+
+This example provides how to apply image segmentation to 3D images using AMD MIGraphX on a given AMD GPU. 
+
+## How to:
+1) User will need to have access to BRATS dataset. Please follow https://www.med.upenn.edu/cbica/brats2019/data.html for how to get the access to dataset.
```
Comment (kahmed10) [inline examples/python_3dunet/README.md:6]:
```suggestion
1) User will need to have access to the BRATS dataset. Please follow https://www.med.upenn.edu/cbica/brats2019/data.html for how to get access to the dataset.
```

File: examples/python_3dunet/visualization_utils.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
+import sys
```
Comment (kahmed10) [inline examples/python_3dunet/visualization_utils.py:1]: the following imports are not being used:
sys
os
pandas
re

File: examples/python_3dunet/3dunet_inference.ipynb [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3930d7b8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import migraphx\n",
```
Comment (kahmed10) [inline examples/python_3dunet/3dunet_inference.ipynb:20]: The following imports are not being used:
PIL
matplotlib.pyplot
time

File: examples/python_3dunet/3dunet_inference.ipynb [hunk 6]
```diff
@@ -0,0 +1,686 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0d061634",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os, sys, glob\n",
```
Comment (kahmed10) [inline examples/python_3dunet/3dunet_inference.ipynb:113]: the following imports are not being used:
sys, glob
pandas
re
matplotlib.pyplot
matplotlib.gridspec
sklearn.utils

File: examples/python_3dunet/3dunet_inference.ipynb [hunk 7]
```diff
@@ -0,0 +1,686 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e734cf64",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
```
Comment (kahmed10) [inline examples/python_3dunet/3dunet_inference.ipynb:133]: We have another cell for imports. Also, only numpy is being used below

**Comment by kahmed10** [comment]
Can the function names and all references to `lable` be renamed to `label`?

File: examples/python_3dunet/3dunet_inference.ipynb [hunk 8]
```diff
@@ -0,0 +1,686 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e3ac8cd3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#np_img = read_image_into_numpy('/code/AMDMIGraphX/bratsdata/MICCAI_BraTS_2019_Data_Training/HGG/BraTS19_TMC_30014_1/resampled')\n",
```
Comment (kahmed10) [inline examples/python_3dunet/3dunet_inference.ipynb:455]: Can you double check this and the other commented out code to see which ones may be useful to keep? And if we are keeping them, I think we should comment on what they do.

File: examples/python_3dunet/3dunet_inference.ipynb (hunk 7 above)
Comment (cagery) [reply examples/python_3dunet/3dunet_inference.ipynb]: thanks for looking at these imports! cleaned them up.

File: examples/python_3dunet/3dunet_inference.ipynb (hunk 8 above)
Comment (cagery) [reply examples/python_3dunet/3dunet_inference.ipynb]: artifact of bringing up. removed.

**Comment by cagery** [comment]
> Can the function names and all references to `lable` be renamed to `label`?

Agreed. Artifact of copy-pasting original post/pre-processing functions. Renamed.

File: examples/python_3dunet/visualization_utils.py (hunk 4 above)
Comment (kahmed10) [reply examples/python_3dunet/visualization_utils.py:1]: Can you please check this file? On github, I don't see it updated.

Comment (cagery) [reply examples/python_3dunet/visualization_utils.py]: oops, missed it, thanks, just updated it

File: examples/python_3dunet/3dunet_inference.ipynb [hunk 9]
```diff
@@ -0,0 +1,643 @@
...
+    "    et = img == 4  # GD-enhancing Tumor (ET)\n",
+    "    \n",
+    "    if out_shape is not None:\n",
+    "        ncr = resize(ncr, out_shape, mode=mode)\n",
+    "        ed = resize(ed, out_shape, mode=mode)\n",
+    "        et = resize(et, out_shape, mode=mode)\n",
+    "    return np.array([ncr, ed, et], dtype=np.uint8)\n",
+    "\n",
+    "np_image=np.zeros((4, 160, 224, 224), dtype=np.float32)\n",
+    "tmp = read_img_sitk('/code/AMDMIGraphX/bratsdata/MICCAI_BraTS_2019_Data_Training/HGG/BraTS19_TMC_30014_1/BraTS19_TMC_30014_1_flair.nii.gz')\n",
```
Comment (turneram) [inline examples/python_3dunet/3dunet_inference.ipynb:378]: It might be a good idea to store the path to the dataset in a variable so that it can be edited in one place. My install of MIGraphX isn't in /code, so even though I created a AMDMIGraphX/bratsdata/ for the dataset, I still had to edit each of these as well as the np.saves below.

Comment (cagery) [reply examples/python_3dunet/3dunet_inference.ipynb]: Created the variable for path. np.save cell is actually not needed or utilized so that's removed. thanks!

**Review by turneram — APPROVED** [review]

**Review by kahmed10 — APPROVED** [review]
